src/ml_board/frontend/Dashboard/src/components/experimentPage/ExperimentDetails/ExperimentDetails.tsx

```diff
@@ -0,0 +1,128 @@
+import { Card, CardContent, Grid } from '@mui/material';
+import moment from 'moment';
+import { useState } from 'react';
+import { AnyKeyValuePairsInterface } from '../ExperimentPage';
+import styles from '../ExperimentPage.module.css';
+import { CardDetails } from './CardDetails';
+
+interface return_obj_interface {
+    [key: string]: string
+}
+
+export function ExperimentDetails({ selectedExperiment }: { selectedExperiment: AnyKeyValuePairsInterface }) {
+
+    const [showMore, setShowMore] = useState(false);
+
+    const conditional_render_exp_overview = () => ({
+        "Experiment ID": selectedExperiment.experiment_id,
+        "Job Type": selectedExperiment.job_type ?? "--",
+        "Device": selectedExperiment.device ?? "--"
+    });
+
+    function conditional_render_exp_state() {
+        const return_obj: return_obj_interface = {};
+
+        if(selectedExperiment.hasOwnProperty("error") && selectedExperiment.error !== null) {
+            return_obj["Job Status"] = "FAILED";
+        }
+        else if(selectedExperiment.job_status === "DONE") {
+            return_obj["Job Status"] = "COMPLETED";
+        }
+        else if(selectedExperiment.job_status === "INIT") {
+            return_obj["Job Status"] = "INITIALIZING";
+        }
+        else if(selectedExperiment.job_status) {
+            return_obj["Job Status"] = selectedExperiment.job_status;
+        }
+        else {
+            return_obj["Job Status"] = "--";
+        }
+
+        return_obj["Model Status"] = selectedExperiment.model_status ?? "--";
+
+        return return_obj;
+    }
```

Comment:
I love all the functions written in this file it's much clearer now, I changed the syntax of the 1st function but it's not necessary so I left the rest, nice work!